### ISSUE TITLE
how to estimate depth ? moderate,far away,close up etc

### ISSUE THREAD

**Issue by akashAD98** [isl-org/MiDaS#168]

**Comment by younesr1** [comment]
I believe that the fundamental nature of monocular depth means that you get relative depth maps not real depth maps in meters. To convert a relative depth map to a real depth map, you'd need the real depth of some part of the image.

**Comment by akashAD98** [comment]
understood, but we can use image disparity to understand depth .if image disparity is very high & object bbox is very high we can say object is near ,same for far away object

**Comment by Mr0cket** [comment]
> I believe that the fundamental nature of monocular depth means that you get relative depth maps not real depth maps in meters. To convert a relative depth map to a real depth map, you'd need the real depth of some part of the image.

Wouldn't it be even the least bit useful if there was a passing reference to this fact in either the documentation, or the paper?

**Comment by akashAD98** [comment]
@dfrumkin yes understood. i build a rule-based system which doing well for predicting its depth.
example. bus,train is big size object .if we detect this in frame & its bbounding box area is very small comapired to total fame size. we can say that image is far away. same for close object -if bbox is very big which is equals to size of frame 

<img width="169" alt="image" src="https://user-images.githubusercontent.com/62583018/174220619-5b6d9f09-2f65-4da3-b44c-79087722e810.png">


<img width="378" alt="image" src="https://user-images.githubusercontent.com/62583018/174220699-8932efbb-4ad5-45b5-b2cb-731a76648273.png">


here i first detected object using yoo framwork & build rule based system,
but i want to combine this semi supervised approch to build this depth estimation

**Comment by dfrumkin** [comment]
@akashAD98 A simplistic approach would be to combine segmentation with depth, compute the average depth for the mask and thus decide how far it is.  The result would still be relative, giving you a bunch of "cardboard" cutouts at various relative depths (may be a good assumption for a standing person, but not for one lying with his feet/head toward the camera).  If you know the real-world size of the object and your camera parameters, you can get some idea of whether it's near or far.

**Comment by duhaitong** [comment]
> @akashAD98 A simplistic approach would be to combine segmentation with depth, compute the average depth for the mask and thus decide how far it is. The result would still be relative, giving you a bunch of "cardboard" cutouts at various relative depths (may be a good assumption for a standing person, but not for one lying with his feet/head toward the camera). If you know the real-world size of the object and your camera parameters, you can get some idea of whether it's near or far.
Hello, I would like to ask a question. I obtained the relative depth through the program and knew the object size and camera internal parameters. Is there any formula for obtaining distance?

**Comment by pragathiacharya** [comment]
> how to get the depth of images ? after getting the result i want to know whether my object is near,far, moderate,very far, what are the parameters we need to use
> 
> ![image](https://user-images.githubusercontent.com/62583018/163805223-17986312-3f68-4ff6-8dc1-7d6572dd3bef.png)

@akashAD98  Have you found the solution for the above posted question ? If so please let me know as i am working on the same. 
Thank  you in advance.
